lib/plugins/aws/deployFunction/index.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+const fs = require('fs');
+const SDK = require('../');
+const validate = require('../lib/validate');
+
+// The Package plugin which is used to zip the service
+const Package = require('../../package');
+
+class AwsDeployFunction {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options || {};
+    this.provider = 'aws';
+    this.sdk = new SDK(serverless);
+
+    this.functionName =
+      `${this.serverless.service.service}-${this.options.stage}-${this.options.function}`;
+
+    this.pkg = new Package(this.serverless, this.options);
```

Comment:
In the future package plugins should be able to listen to generic lifecycle events. At the moment they are namespaced to specific commands which makes it impossible to simply add packaging to anything.